sdk/go/pulumi/stack_reference.go

```diff
@@ -39,6 +39,24 @@ func (s *StackReference) GetIDOutput(name StringInput) IDOutput {
 	}).(IDOutput)
 }
 
+// GetFloat64Output returns a stack output keyed by the given name as an Float64Output
+func (s *StackReference) GetFloat64Output(name StringInput) Float64Output {
+	return s.GetOutput(name).ApplyT(func(out interface{}) float64 {
+		var res float64
+		if out != nil {
+			res = out.(float64)
+		}
+		return res
+	}).(Float64Output)
+}
+
+// GetIntOutput returns a stack output keyed by the given name as an IntOutput
+func (s *StackReference) GetIntOutput(name StringInput) IntOutput {
+	return s.GetFloat64Output(name).ApplyT(func(out float64) int {
+		return int(out)
```

Comment:
Similar to the feedback above, we should catch and propagate the error here. In addition, we should just use `strconv.Atoi` directly rather than delegating to `s.GetFloat64Output`: 

https://pkg.go.dev/strconv#example-Atoi

Reason being that in the error case, the user who is trying to get an `int` will get an error message complaining that the value could not be parsed as a float64 which would be pretty confusing.

Reply:
Now I understand the error catch and propagation.
I modified the code to process the type conversion within the function not delegating to `s.GetFloat64Output` and return error when failed.
Please check it.